src/server.c

```diff
@@ -641,20 +641,19 @@ int htNeedsResize(dict *dict) {
  * In non cluster-enabled setup, it resize main/expires dictionary based on the same condition described above. */
 void tryResizeHashTables(int dbid) {
     redisDb *db = &server.db[dbid];
-    int slot = 0;
     for (dbKeyType subdict = DB_MAIN; subdict <= DB_EXPIRES; subdict++) {
-        dbIterator *dbit = dbIteratorInitFromSlot(db, subdict, db->sub_dict[subdict].resize_cursor);
-        for (int i = 0; i < CRON_DBS_PER_CALL; i++) {
-            dict *d = dbGetDictFromIterator(dbit);
-            slot = dbIteratorGetCurrentSlot(dbit);
-            dbIteratorNextDict(dbit);
-            if (!d) break;
```

Comment:
i wonder why we needed this check and don't anymore.
is there something we're missing?

Reply:
In fact, this check will never take effect, which is also the reason why it keeps blocking on the last non-empty slot. And now we use `resize_cursor` being -1 as the termination condition.

Reply:
but can't it be that a dict (resize_cursor) was not empty when we left the loop last time, and when we came back it is empty?

Reply:
yes, but make an additional attempt on an empty slot is not a big deal?

Reply:
question is if the dict could be NULL leading to a crash...

Reply:
I know, but now all dicts are initialized as non-NULL, and if we want to change this design, it will not only cause issues in this instance... Do we have any plans to only create dictionaries that are held by the current node?